src/civ2.js

```diff
@@ -238,89 +216,193 @@ module.exports = function(robot) {
+    return async (msg) => {
       try {
-        parsedPrs.set(branch, Date.now());
-        await civ2.commentPtReferences(branch);
-        cleanParsedPrs();
-        return res.send('OK');
+        await func(msg);
       } catch (err) {
-        console.error(err);
-        robot.messageRoom(room, `An error occured while looking for PT references in "${branch}": ${err}`);
-        res.status(500).send('Error');
+        respondToError(err, msg, respondMethod);
       }
-    }
-  });
-
-  robot.router.post('/hubot/civ2/create-pr', routes.createPr);
-  robot.router.post('/hubot/civ2/announce-pr', routes.announcePRs);
+    };
+  }
 };
 
-const MAX_PRDATE = 1000 * 60 * 5;
+// -----------------------------------------------------------------------------
+// PR MERGE HANDLERS
+// -----------------------------------------------------------------------------
 
-function cleanParsedPrs() {
-  const now = Date.now();
-  for (const pr of parsedPrs.keys()) {
-    const prDate = parsedPrs.get(pr);
-    if (now - prDate > MAX_PRDATE) {
-      parsedPrs.delete(pr);
+function inferGHPrAction(pr) {
+  const { action, merged } = pr;
+  return action === 'closed' && merged === true ? 'merged' : action;
+}
+
+async function tryDestryoFeatureCluster(branch, robot, res) {
+  const destroyFn = () => civ2.destroyFeatureCluster(branch);
+  const destroyErrMsg = `triggering destruction of feature cluster "${branch}"`;
+  return tryAndNotifyErr(destroyFn, destroyErrMsg, robot, res);
+}
+
+async function tryDeleteBranch(repo, branch, base, robot, res) {
+  const deleteFn = async () => {
+    await civ2.deleteBranch(repo, branch);
+    const message = `<https://github/com/sutoiku/${repo}/branches|Branch ${branch}> of <https://github/com/sutoiku/${repo}|${repo}> was merged into ${base}, I deleted it.`;
+    robot.messageRoom(CHANNELS.default, message);
+  };
+
+  return tryAndNotifyErr(deleteFn, `deleting branch "${branch}"`, robot, res);
+}
+
+async function announcePrMerge(branch, robot, base, mergedPRs, res) {
+  if (mergedPRs.has(branch)) {
+    return;
+  }
+
+  const icon = inferIconFromBranchName(branch);
+  const text = [`${icon}Branch \`${branch}\` was merged into \`${base}\`.`];
+  try {
+    mergedPRs.set(branch, { when: Date.now() });
+
+    const ptDescription = await generateBranchPtDescription(branch);
+    if (ptDescription) {
+      text.push(ptDescription);
     }
+
+    robot.messageRoom(CHANNELS.releaseCandidates, text.join('\n'));
+    cleanPrMap(mergedPRs);
+    return true;
+  } catch (err) {
+    error(err);
+    console.log(err);
+    robot.messageRoom(CHANNELS.default, `An error occured while announcing PR merge (${branch}) : ${err}`);
+    res.status(500).send('Error');
+  }
+}
+
+function inferIconFromBranchName(branch) {
+  const kind = branch.split('/')[0];
+  switch (kind) {
+    case 'feature':
+      return ':gift: ';
+    case 'fix':
+      return ':ladybug: ';
+    case 'bug':
+      return ':ladybug: ';
+    case 'chore':
```

Comment:
Add `refactor` as alias

Reply:
Don't be shy, do it in your first bot commit :)